### PR TITLE
Require pagination in API queries

### LINKED ISSUE #3301: Require providing "first" or "last" arguments in connection fields
Connection fields should require passing `first` or `last` arguments in order to always use pagination and avoid potentially expensive queries fetching whole collections. Graphene provides a setting that enables that requirement: `RELAY_CONNECTION_ENFORCE_FIRST_OR_LAST`.

### PR DESCRIPTION
**Pull request by maarcingebala**
Fixes #3301 

- [x] Require pagination in API
- [x] Allow querying `totalCount` without pagination
- [x] Adjust tests
- [x] Update dashboard queries 

### Pull Request Checklist

1. [x] Privileged views and APIs are guarded by proper permission checks.
1. [x] All visible strings are translated with proper context.
1. [x] All data-formatting is locale-aware (dates, numbers, and so on).
1. [x] Database queries are optimized and the number of queries is constant.
1. [x] The changes are tested.
1. [ ] The code is documented (docstrings, project documentation).
1. [x] GraphQL schema and type definitions are up to date.

### CODE REVIEW

**Comment by maarcingebala** [comment]
Todo in Dashboard 2.0 UI (can be done in a followup PR):
- [ ] add pagination in the categories list view
- [ ] add pagination of subcategories in the category details view
- [ ] add autocomplete support in the product create/edit views for the following fields: category, collection, product type.

File: tests/api/test_core.py [hunk 1]
```diff
@@ -104,3 +105,36 @@ def test_reporting_period_to_date():
...
+    query = """
+    query {
+        products {
+            totalCount
+        }
+    }
+    """
+    response = api_client.post_graphql(query)
+    content = get_graphql_content(response)
+    assert content['data']['products']['totalCount'] == 1
```
Comment (akjanik) [inline tests/api/test_core.py]: I would avoid hardcoding values.
```suggestion
    assert content['data']['products']['totalCount'] == Product.objects.count()
```

File: saleor/graphql/account/types.py [hunk 2]
```diff
@@ -41,20 +42,28 @@ def resolve_country(self, info):
 
 
 class User(CountableDjangoObjectType):
-    permissions = graphene.List(
-        PermissionDisplay, description='List of user\'s permissions.')
     addresses = gql_optimizer.field(
-        graphene.List(
-            Address, description='List of all user\'s addresses.'),
+        graphene.List(Address, description='List of all user\'s addresses.'),
```
Comment (NyanKiyoshi) [inline saleor/graphql/account/types.py]: > of all

That sounds weird/ wrong, don't you think?
Maybe we could say that:
```suggestion
        graphene.List(Address, description='The address list of the user.'),
```

File: saleor/graphql/account/types.py [hunk 3]
```diff
@@ -41,20 +42,28 @@ def resolve_country(self, info):
...
         model_field='addresses')
     note = graphene.String(description='A note about the customer')
+    orders = gql_optimizer.field(
+        PrefetchingConnectionField(
+            'saleor.graphql.order.types.Order',
+            description='List of user\'s orders.'),
```
Comment (NyanKiyoshi) [inline saleor/graphql/account/types.py]:
```suggestion
            description='The order list of the user.'),
```

File: saleor/graphql/account/types.py [hunk 4]
```diff
@@ -41,20 +42,28 @@ def resolve_country(self, info):
...
+        model_field='orders')
+    permissions = graphene.List(
+        PermissionDisplay, description='List of user\'s permissions.')
```
Comment (NyanKiyoshi) [inline saleor/graphql/account/types.py]:
```suggestion
        PermissionDisplay, description='The permission list of the user.')
```

Comment (maarcingebala) [reply saleor/graphql/account/types.py]: I'd leave the descriptions unchanged for now. They need to be retaken anyway in a dedicated PR probably. Also, I think that saying "List of something.." is quite common in our API.

**Pull request merged**